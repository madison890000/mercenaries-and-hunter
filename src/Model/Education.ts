import DataModel from './types';
import Base from './Base';

interface IEducation {
    college: string;
    major: string;
    degree: DataModel.Degree;
    start: Date;
    end: Date;
}

export default class Education extends Base {
    public college: string;
    public major: string;
    public degree: DataModel.Degree;
    public start: Date;
    public end: Date;

    constructor({ college, major, degree, start, end }: IEducation) {
        super();
        this.major = major;
        this.college = college;
        this.degree = degree;
        this.start = start;
        this.end = end;
    }
}
