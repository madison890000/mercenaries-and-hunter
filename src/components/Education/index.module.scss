.education {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.times {
  & > span {
    display: inline-block;
    margin: 0 8px;
    min-width: 90px;
    text-align: right;
    font-style: italic;
    font-size: 18px;
  }
}

.major {
  font-style: italic;
  color: gray;
  margin-left: 12px;
  font-size: 16px;
}
