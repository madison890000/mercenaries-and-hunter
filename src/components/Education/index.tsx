import React from 'react';
import styles from './index.module.scss';
import DataModel from '../../models/types';
import Times from '../Times';
import Tag from "../Tag";

interface EducationProps {
    college: string;
    degree: DataModel.Degree;
    major: string;
    start: Date;
    end: Date;
}

const Education = ({start, end, college, degree, major}: EducationProps) => {
    return (
        <div className={styles.education}>
            <div>
                <span>{college}</span>
            </div>
            <div>
                <Tag type="filled">{major}</Tag>
                <Tag type="less">{degree}</Tag>
            </div>
            <Times start={start} end={end}/>
        </div>
    );
};

export default Education;
