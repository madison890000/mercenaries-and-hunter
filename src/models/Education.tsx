import React from "react";
import Base from './Base';
import EditText from './EditText';
import Times from './Times';
import {Tag} from "./components";
import styled from "styled-components";

export interface IEducation {
    college: string;
    major: string;
    degree: string;
    start: string;
    end: string;
}

const EducationUI = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  font-size: var(--base-font-size-middle);
`

class Education extends Base {
    public college: EditText;
    public major: EditText;
    public degree: EditText;
    private times: Times;


    constructor({college, major, degree, start, end}: IEducation) {
        super();
        this.major = new EditText(major).setParent(this);
        this.college = new EditText(college).setParent(this);
        this.degree = new EditText(degree).setParent(this);
        this.times = new Times(start, end).setParent(this);
    }

    View = () => {
        return (
            <EducationUI>
                <div>
                    <this.college.Show/>
                </div>
                <div>
                    <Tag type="filled">
                        <this.major.Show/>
                    </Tag>
                    <Tag type="less">
                        <this.degree.Show/>
                    </Tag>
                </div>
                <this.times.Show/>
            </EducationUI>
        )
    }

    Edit = () => {
        return (
            <EducationUI>
                <div>
                    <this.college.Show/>
                </div>
                <div>
                    <this.major.Show/>
                    <this.degree.Show/>
                </div>
                <this.times.Show/>
            </EducationUI>
        )
    }
}

export default Education