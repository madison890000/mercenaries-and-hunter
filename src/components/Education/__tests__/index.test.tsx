import React from 'react';
import Education from '../index';
import renderer from 'react-test-renderer';
import DataModel from '../../../models/types';

test('renders Education Bachelor', () => {
    const start = new Date('2012-01-01');
    const end = new Date('2013-01-01');
    const EducationDom = renderer.create(<Education start={start} end={end} college="test college" degree={DataModel.Degree.Bachelor} major="test major" />).toJSON();
    expect(EducationDom).toMatchSnapshot();
});
test('renders Education Master', () => {
    const start = new Date('2012-01-01');
    const end = new Date('2013-01-01');
    const EducationDom = renderer.create(<Education start={start} end={end} college="test college" degree={DataModel.Degree.Master} major="test major" />).toJSON();
    expect(EducationDom).toMatchSnapshot();
});
test('renders Education PhD', () => {
    const start = new Date('2012-01-01');
    const end = new Date('2013-01-01');
    const EducationDom = renderer.create(<Education start={start} end={end} college="test college" degree={DataModel.Degree.PhD} major="test major" />).toJSON();
    expect(EducationDom).toMatchSnapshot();
});
