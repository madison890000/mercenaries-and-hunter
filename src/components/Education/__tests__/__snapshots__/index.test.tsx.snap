// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Education Bachelor 1`] = `
<div
  className="education"
>
  <div>
    <span>
      test college
    </span>
  </div>
  <div>
    <span>
      test major
    </span>
    <span
      className="major"
    >
      (
      BACHELOR
      )
    </span>
  </div>
  <div
    className="times"
  >
    <span>
      Jan, 2012
    </span>
    <span
      className="divider"
    />
    <span>
      Jan, 2013
    </span>
  </div>
</div>
`;

exports[`renders Education Master 1`] = `
<div
  className="education"
>
  <div>
    <span>
      test college
    </span>
  </div>
  <div>
    <span>
      test major
    </span>
    <span
      className="major"
    >
      (
      MASTER
      )
    </span>
  </div>
  <div
    className="times"
  >
    <span>
      Jan, 2012
    </span>
    <span
      className="divider"
    />
    <span>
      Jan, 2013
    </span>
  </div>
</div>
`;

exports[`renders Education PhD 1`] = `
<div
  className="education"
>
  <div>
    <span>
      test college
    </span>
  </div>
  <div>
    <span>
      test major
    </span>
    <span
      className="major"
    >
      (
      PHD
      )
    </span>
  </div>
  <div
    className="times"
  >
    <span>
      Jan, 2012
    </span>
    <span
      className="divider"
    />
    <span>
      Jan, 2013
    </span>
  </div>
</div>
`;
